### ISSUE TITLE
Invalid ES6 module rewrite

### ISSUE THREAD

**Issue by cramforce** [google/closure-compiler#1830]
I was just trying to upgrade our compiler to head (To test out https://github.com/google/closure-compiler/commit/96dbbf6820d1e9ae862cf91a83116c391ba08519) and ran into the following issue:

The compiler throws an error on the first global `let` it sees in the first module it compiles.

Error: `The nameNode for Var ModeDef$$module$src$mode must be a descendant of one of: [118, 162, 149, 105, 158, 120, 83]`

This is triggered by the first line in a file that is just: `let mode;`

More info:

```
Command failed: java -jar -XX:+TieredCompilation build-system/runner/dist/runner.jar --flagfile="/var/folders/jk/j0ymfl3j6nj3w3k6_ngj7gfh003p7j/T/b57bfc05-be08-4697-ba20-9ea6a1d88227" --compilation_level=SIMPLE_OPTIMIZATIONS --assume_function_wrapper=true --language_in=ECMASCRIPT6 --language_out=ECMASCRIPT5 --externs=build-system/amp.extern.js --js_module_root=node_modules/ --js_module_root=build/fake-module/ --entry_point=./3p/integration.js --only_closure_dependencies=true --output_wrapper=(function(){var process={env:{NODE_ENV:"production"}};%output%})();
//# sourceMappingURL=f.js.map
 --create_source_map=build/cc/_3p_integration.js.map --source_map_location_mapping=|http://localhost:8000/ --warning_level=DEFAULT --js_output_file="build/cc/_3p_integration.js"
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

The nameNode for Var ModeDef$$module$src$mode must be a descendant of one of: [118, 162, 149, 105, 158, 120, 83]
  Node(BLOCK): [source unknown]
  Parent(BLOCK): [source unknown]

	at com.google.javascript.jscomp.Var.declarationType(Var.java:225)
```

Does anyone have an idea what is going on?

**Comment by cramforce** [comment]
I, uh, recompiled (adding some debug info) and the problem went away.

Closing for now.
